File: csharp/isogram/Isogram.cs
using System;

public static class Isogram
{
    public static bool IsIsogram(string word)
    {
        string wordChars = "";

        foreach (char letter in word)
        {
            char value = char.ToUpper(letter);
            if (value < 65 || value > 90)
                continue;
            
            if (wordChars.Contains(value))
                return false;
            
            wordChars += value;
        }
        return true;
    }
}

File: c/isogram/src/isogram.c
#include "isogram.h"
#include <string.h>
#include <ctype.h>

bool is_isogram(const char phrase[])
{
    if (phrase == NULL)
        return false;
    else if (strlen(phrase) == 0)
        return true;
    
    bool charactersPresent[26] = { false };

    for (int i = 0; i < (int) strlen(phrase); ++i)
    {
        char letter = tolower(phrase[i]);

        if (letter >= 'a' && letter <= 'z')
        {
            if (charactersPresent[letter - 'a'] == false)
                charactersPresent[letter - 'a'] = true;
            else
                return false;
        }
    }
    return true;
}

